components/st-microelectronics/stm-api.stanza

```diff
@@ -142,25 +143,34 @@ public defn set-boot (stm:JITXObject, boot-from:String) :
     if has-pin?(stm, "BOOT[1]") : 
       check connected(stm.BOOT[1])
 
-defn supports? (module:InstantiableType, bundle:PortType) -> True|False :
+defn supports? (module:InstantiableType, bundle:Bundle) -> True|False :
   pcb-module temp :
     inst i : module
     require r:bundle from i
   set-main-module(temp) ;uh oh
   val result = assign-pins?()
   result is-not False
 
-public defn connect-debug (stm:JITXObject, debug-interface:PortType, connector:Instantiable) :
+public defn connect-debug (stm:JITXObject, debug-interface:Bundle, connector:Instantiable) :
   inside pcb-module:
     if supports?(instantiable-type(stm), debug-interface) and supports?(connector, debug-interface):
-      inst debug-con : connector
+      public inst debug-con : connector
       require debug-m:debug-interface from stm
       require debug-c:debug-interface from debug-con
+      
       net DEBUG (debug-m debug-c)
       schematic-group(debug-con) = debug
+
       if has-port?(debug-con, "power") :
```

Comment:
This is fragile: are there cases where `power` is **not** required from the connector?

Reply:
It is fragile, and current behavior is not desired -- failing to find the port and just not making the connection.

It is possible that power comes from somewhere else and not the debug connector, but let's add a fatal here for the time being if the connection is not found on the debug connector.